sycl/test-e2e/Config/device_selector.cpp

```diff
@@ -8,20 +8,19 @@
 
 #include <iostream>
 
-class RejectEverything : public sycl::device_selector {
-public:
-  int operator()(const sycl::device &Device) const final {
-    // Negative value means that a device must not be selected
-    return -1;
-  }
-};
-
 int main() {
-  RejectEverything Selector;
+  auto RejectEverything = [](const sycl::device &) { return -1; };
   try {
-    sycl::device Device(Selector);
-  } catch (sycl::runtime_error &E) {
-    return 0;
+    sycl::device Device(RejectEverything);
+  } catch (sycl::exception &E) {
+    if (E.code() == sycl::errc::runtime &&
+        std::string(E.what()).find("No device of requested type available. -1 "
+                                   "(PI_ERROR_DEVICE_NOT_FOUND)") !=
```

Comment:
I'm updating SYCL exception to align with SYCL 2020 spec and as part of it I'm going to remove the `PI_ERROR_DEVICE_NOT_FOUND` part of the message. The reason is `PI_*` are internals of our implementation and not something standard/common in the industry. The closed thing would be OpenCL/L0 error codes, but those are different (logically, not essentially bitwise different) from PI/UR error codes. Backend error codes could be accessed from the `sycl::exception` in a backend specific way defined by the corresponding interop spec, but I don't think it's worth checking that in this test at all. The first part of the condition (`errc::runtime`) and possibly "No device of request type available." substring should suffice.